### ISSUE TITLE
ViT generates different output in successive runs with same input

### ISSUE THREAD

**Issue by andretag** [kornia/kornia#1693]
### Describe the bug

Is it expected that if I pass the same image as input of the same transformer, the output can be different? This happens unless I call `torch.manual_seed(SEED)` before calling the transformer. 

### Reproduction steps

```bash
import kornia
import torch

set_seed = False
transformer = kornia.contrib.VisionTransformer(image_size=64, patch_size=16, in_channels=1, embed_dim=64, depth=4, num_heads=4, dropout_rate=0.1, dropout_attn=0.1, backbone=None)
single_in_imgs = torch.rand((1, 1, 64, 64))
if set_seed is True: torch.manual_seed(0)
out1 = transformer(single_in_imgs)
if set_seed is True: torch.manual_seed(0)
out2 = transformer(single_in_imgs)
assert torch.all(torch.isclose(out1[0, :], out2[0, :])), "Transformer output is different"
```

### Expected behavior

I would expect the output to be the same, given the same input, even if I do not manually set the torch random seed. 

### Environment

```shell
PyTorch version: 1.8.2+cu111
Is debug build: False
CUDA used to build PyTorch: 11.1
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.3 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.16.3
Libc version: glibc-2.31

Python version: 3.8.10 (default, Mar 15 2022, 12:22:08)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-5.13.0-40-generic-x86_64-with-glibc2.29
Is CUDA available: True
CUDA runtime version: 11.1.105
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3090
Nvidia driver version: 470.103.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.22.2
[pip3] torch==1.8.2+cu111
[pip3] torch-tb-profiler==0.4.0
[pip3] torchaudio==0.8.2
[pip3] torchvision==0.9.2+cu111
[conda] Could not collect
```

**Comment by carbocation** [comment]
For what it's worth, I have observed the same thing with other ViT implementations like https://github.com/lucidrains/vit-pytorch . (I have double checked that the model is in eval mode with torch.no_grad, too.)

**Comment by andretag** [comment]
I am wondering if this is not a but, but simply the behavior of the dropout layers
